packages/@aws-cdk/aws-apigateway/lib/restapi.ts

```diff
@@ -426,6 +436,45 @@ export class RestApi extends Resource implements IRestApi {
 
     resource.node.addDependency(apiResource);
   }
+
+  private configureEndpointConfiguration(props: RestApiProps): CfnRestApi.EndpointConfigurationProperty | undefined {
+    if (props.endpointTypes && props.endpointConfiguration) {
+      throw new Error('Only one of the RestApi props, endpointTypes or endpointConfiguration, is allowed');
+    }
+    if (props.endpointConfiguration) {
+      const vpcEndpointIds = (props.endpointConfiguration.vpcEndpoints) ?
+        props.endpointConfiguration.vpcEndpoints.map(vpcEndpoint => vpcEndpoint.vpcEndpointId) : undefined;
```

Comment:
Would [optional chaining](https://www.typescriptlang.org/docs/handbook/release-notes/overview.html#optional-chaining) simplify this?

```suggestion
      const vpcEndpointIds = props.endpointConfiguration.vpcEndpoints?.map(vpce => vpce.vpcEndpointId);
```